src/containers/CreatePipelineRun/YAMLEditor.js

```diff
@@ -0,0 +1,245 @@
+/*
+Copyright 2022 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { useIntl } from 'react-intl';
+import { ALL_NAMESPACES, urls } from '@tektoncd/dashboard-utils';
+import {
+  Button,
+  Form,
+  FormGroup,
+  InlineNotification
+} from 'carbon-components-react';
+import yaml from 'js-yaml';
+import React, { useState } from 'react';
+import CodeMirror from '@uiw/react-codemirror';
+import { StreamLanguage } from '@codemirror/language';
+import { yaml as cmyaml } from '@codemirror/legacy-modes/mode/yaml';
+import { useNavigate } from 'react-router-dom-v5-compat';
+import { createPipelineRunRaw, useSelectedNamespace } from '../../api';
+
+const initialState = {
+  creating: false,
+  pipelineError: false,
+  submitError: '',
+  validationError: false,
```

Comment:
This seems redundant. Can we remove this and just use the presence of a `validationErrorMessage` to determine if there's something to be shown?